src/integrations/gitlens/gitlensImpl.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { commands, Disposable, extensions } from 'vscode';
+import { CreatePullRequestActionContext, GitLensApi } from './gitlens';
+
+export class GitLensIntegration implements Disposable {
+	private _extensionsDisposable: Disposable;
+	private _subscriptions: Disposable[] = [];
+
+	constructor() {
+		this._extensionsDisposable = extensions.onDidChange(this.onExtensionsChanged, this);
+		this.onExtensionsChanged();
+	}
+
+	dispose() {
+		this._extensionsDisposable.dispose();
+		Disposable.from(...this._subscriptions).dispose();
+	}
+
+	private register(api: GitLensApi) {
+		this._subscriptions.push(
+			api.registerActionRunner('createPullRequest', {
+				name: 'GitHub Pull Requests and Issues',
+				label: 'Create Pull Request',
+				run: function (context: CreatePullRequestActionContext) {
+					// For now only work with branches that aren't remote and have an upstream
+					if (context.branch.isRemote || !context.branch.upstream) {
+						return;
+					}
+
+					commands.executeCommand('pr.create', { repoPath: context.repoPath, compareBranch: context.branch.name });
+				}
+			}),
+			// api.registerActionRunner('openPullRequest', {
```

Comment:
remove this for now?

Reply:
Yeah, I'm planning on finishing it before we merge